components/molecules/table/components/pagination.js

```diff
@@ -8,28 +8,74 @@ import { array, func, number } from 'prop-types'
 
 // UI
 import { Pagination } from '../../../'
+import { Row } from '../../../atoms/'
+import Select from 'react-select'
 
+// Style
+import styled from 'styled-components'
+
+const Items = [
+  { label: '25', value: 25 },
+  { label: '50', value: 50 },
+  { label: '100', value: 100 },
+  { label: 'All', value: Infinity }
+]
 export const TablePagination = memo(
-  ({ currentPage, handlePagination, pageCount, perPage, rows }) => (
-    <Pagination
-      currentPage={currentPage}
-      onPageChange={handlePagination}
-      pageCount={pageCount}
-      showNextAndPrev
-      size="sm"
-    />
-  )
-)
+  ({
+    currentPage,
+    handlePagination,
+    handlePaginationSize,
+    paginationSize,
+    pageCount,
+    perPage,
+    rows
+  }) => {
+    const pagineSizeChangeHandler = (size) => {
+      handlePaginationSize(size.value)
+    }
 
+    return (
+      <Row>
+        <StyledDivision width={paginationSize ? '70%' : '100%'}>
+          <Pagination
+            currentPage={currentPage}
+            onPageChange={handlePagination}
+            pageCount={pageCount}
+            showNextAndPrev
+            size="sm"
+          />
+        </StyledDivision>
+        {paginationSize && (
```

Comment:
@shahabsadeghi We are trying to create new files for the individual components, this would take the handlers style and code to a new files so keeping it clean - and also helping react to reduce renders if necessary - do plan out structure carefully